src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -0,0 +1,129 @@
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME // Hide confidential client on mobile platforms
+
+    /// <summary>
+    /// </summary>
+    public class ConfidentialClientApplicationBuilder : AbstractApplicationBuilder<ConfidentialClientApplicationBuilder>
+    {
+        /// <inheritdoc />
+        internal ConfidentialClientApplicationBuilder(ApplicationConfiguration configuration)
+            : base(configuration)
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public static ConfidentialClientApplicationBuilder CreateWithApplicationOptions(
+            ConfidentialClientApplicationOptions options)
+        {
+            var config = new ApplicationConfiguration();
+            return new ConfidentialClientApplicationBuilder(config)
+                   .WithOptions(options).WithClientCredential(new ClientCredential(options.ClientCredentialSecret));
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="clientId"></param>
+        /// <returns></returns>
+        public static ConfidentialClientApplicationBuilder Create(string clientId)
+        {
+            var config = new ApplicationConfiguration();
+            return new ConfidentialClientApplicationBuilder(config).WithClientId(clientId);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="clientId"></param>
+        /// <param name="authority"></param>
+        /// <param name="validateAuthority"></param>
+        /// <returns></returns>
+        public static ConfidentialClientApplicationBuilder Create(
+            string clientId,
+            string authority,
+            bool validateAuthority = true)
+        {
+            return Create(clientId).WithAuthority(authority, validateAuthority, true);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="tokenCache"></param>
+        /// <returns></returns>
+        public ConfidentialClientApplicationBuilder WithAppTokenCache(ITokenCache tokenCache)
+        {
+            Config.AppTokenCache = tokenCache;
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="tokenCache"></param>
+        /// <returns></returns>
+        public ConfidentialClientApplicationBuilder WithUserTokenCache(ITokenCache tokenCache)
+        {
+            Config.UserTokenCache = tokenCache;
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="clientCredential"></param>
+        /// <returns></returns>
+        public ConfidentialClientApplicationBuilder WithClientCredential(ClientCredential clientCredential)
```

Comment:
Actually as we are flattening everything. I wonder if we should not have:
.WithApplicationSecret(string - or secure string?)
.WithCertificate(certificate)
Do we need ClientCredential?